salt/_pillar/metalk8s_nodes.py

```diff
@@ -40,6 +40,28 @@ def node_info(node, ca_minion):
     if node["metadata"]["name"] == ca_minion:
         roles.add("ca")
 
+    # Introduced by https://scality.atlassian.net/browse/ARTESCA-10079
+    # as a workaround in cases which bootstrap setup fails after the
+    # kubernetes API server is up and running but node role labels are
+    # still not set.
+    #
+    # The issue is that when the API server is running the ext_pillar
+    # will overwrite the hardcoded roles with the ones from the API.
+    # It will cause bootstrap setup failure as described here:
+    # https://github.com/scality/metalk8s/issues/2137
+    #
+    # The `is_bootstrap` flag is set in pillar only when run the
+    # `metalk8s.orchestrate.bootstrap` state.
+    if pillar.get("is_bootstrap", False):
+        log.debug(
+            "Workaround: Adding bootstrap roles to pillar for node %s",
+            node["metadata"]["name"],
+        )
+        roles.add("bootstrap")
+        roles.add("master")
+        roles.add("etcd")
```

Comment:
I would do it before the `if` above and only if `roles` is empty and fill it with what is already in the `pillar` instead of hardcoding roles here

Reply:
Fixed !!!

Reply:
Fixed.